wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -515,42 +514,29 @@ public final class MLSService: MLSServiceInterface {
         }
     }
 
-    private func claimKeyPackages(for users: [MLSUser]) async throws -> [KeyPackage] {
-        logger.info("claiming key packages for users: \(users)")
-        do {
-            guard let context = context else { return [] }
+    private func claimKeyPackages(
+        for users: [MLSUser]
+    ) async -> [KeyPackage] {
+
+        guard let context = context else { return [] }
 
-            var result = [KeyPackage]()
+        var result = [KeyPackage]()
 
-            for try await keyPackages in claimKeyPackages(for: users, in: context) {
+        for user in users {
+            do {
+                let keyPackages = try await actionsProvider.claimKeyPackages(
+                    userID: user.id,
+                    domain: user.domain,
+                    excludedSelfClientID: user.selfClientID,
+                    in: context.notificationContext
+                )
                 result.append(contentsOf: keyPackages)
+            } catch {
+                logger.warn("failed to claim key packages for user (\(user.id)): \(String(describing: error))")
```

Comment:
If i'm thinking ahead about how will want to insert a system message informing about which users's did not get added, I don't think we can swallow the error. 

Rather I think will need a more detailed error which includes which users we couldn't claim key packages for:
`failedToClaimKeyPackages(users: [MLSUser])`

The caller could then exclude these users and an retry the operation and then inserted a system mesasge with the excluded users.

Reply:
Logic has been updated